session-helper/flatpak-session-helper.c

```diff
@@ -33,6 +33,7 @@
 static char *monitor_dir;
 
 static GHashTable *client_pid_data_hash = NULL;
+static GHashTable *file_monitor_hash = NULL;
```

Comment:
`file_monitor_hash` is redundant, since you only ever insert into it, and never lookup or remove. Given that the number of calls to `setup_file_monitor()` is bounded, you might as well just return the `MonitorData` from `setup_file_monitor()`, and assign it to a `g_autoptr(MonitorData)` variable in the caller.

That would resolve the second point I was going to make about this hash table, which is that it never gets freed. (I realise it gets freed on exit from the program, but that makes it harder to use valgrind to detect actual leaks.)

Reply:
Done. Although... by inspection there are a lot of things in this program which knowingly leak, but nothing which isn't expected to persist for the life of the program.